### PR TITLE
fix(metrics): remove resources that aren't updated

### PR DESCRIPTION
**Pull request by jpeeler**
These resources are updated in the catalog operator, so a new endpoint
needs to be created to scrape them from there instead. This commit
merely removes them from being served from the olm operator.

~(So this is really just a partial fix.)~ Updated to actually serve the new metrics properly.

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -129,15 +129,16 @@ var (
...
 		},
 	)
 )
 
 func Register() {
 	prometheus.MustRegister(csvCount)
-	prometheus.MustRegister(installPlanCount)
-	prometheus.MustRegister(subscriptionCount)
-	prometheus.MustRegister(catalogSourceCount)
+	//TODO: these metrics are actually handled in the catalog operator
```
Comment (alecmerdler) [inline pkg/metrics/metrics.go]: Can we delete these lines instead of commenting them out?

Comment (jpeeler) [reply pkg/metrics/metrics.go]: You're right, I can do better.

File: cmd/olm/main.go [hunk 2]
```diff
@@ -100,12 +100,11 @@ func main() {
...
 		w.WriteHeader(http.StatusOK)
 	})
-	// TODO: both of the following require vendor updates (add k8s.io/apiserver and update prometheus)
-	//healthz.InstallHandler(mux) //(less code)
-	//mux.Handle("/metrics", promhttp.Handler()) //other form is deprecated
-	http.Handle("/metrics", prometheus.Handler())
 	go http.ListenAndServe(":8080", nil)
 
+	http.Handle("/metrics", promhttp.Handler())
+	go http.ListenAndServe(":8081", nil)
```
Comment (ecordell) [inline cmd/olm/main.go]: Also serving a nil handler here

File: cmd/catalog/main.go [hunk 3]
```diff
@@ -87,6 +93,10 @@ func main() {
 		log.Panicf("error configuring operator: %s", err.Error())
 	}
 
+	http.Handle("/metrics", promhttp.Handler())
+	go http.ListenAndServe(":8081", nil)
```
Comment (ecordell) [inline cmd/catalog/main.go]: This is serving a nil handler - did you mean to serve metrics on this port?

File: cmd/olm/main.go [hunk 4]
```diff
@@ -100,12 +100,11 @@ func main() {
 	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
 	})
-	// TODO: both of the following require vendor updates (add k8s.io/apiserver and update prometheus)
-	//healthz.InstallHandler(mux) //(less code)
-	//mux.Handle("/metrics", promhttp.Handler()) //other form is deprecated
-	http.Handle("/metrics", prometheus.Handler())
 	go http.ListenAndServe(":8080", nil)
```
Comment (ecordell) [inline cmd/olm/main.go]: Similar issue here, though this has clearly been here a while

File: cmd/olm/main.go (hunk 2 above)
Comment (jpeeler) [reply cmd/olm/main.go]: Yes, nil is intentional, it should use the "DefaultServeMux".

File: cmd/catalog/main.go (hunk 3 above)
Comment (jpeeler) [reply cmd/catalog/main.go]: This port change was intentional, though perhaps it's not the best? OLM serves healthz from port 8080, so I wanted the metrics ports to be consistent.

File: cmd/olm/main.go (hunk 4 above)
Comment (jpeeler) [reply cmd/olm/main.go]: See other comments, I'm nearly certain I've confirmed that healthz was being served correctly.

File: cmd/olm/main.go (hunk 2 above)
Comment (ecordell) [reply cmd/olm/main.go]: Gotcha - didn't know about DefaultServeMux - I think go didn't have a muxer last time I used `http`? 😂

**Comment by ecordell** [comment]
@jpeeler I'm optimistic about this merging if you rebase?

**Comment by jpeeler** [comment]
/skip gitlab-ci/pipeline
not sure if this will work...

**Pull request merged**